staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -909,17 +911,27 @@ func (s *store) getState(ctx context.Context, getResp *clientv3.GetResponse, key
 			return nil, err
 		}
 	} else {
-		data, stale, err := s.transformer.TransformFromStorage(ctx, getResp.Kvs[0].Value, authenticatedDataString(key))
-		if err != nil {
-			return nil, storage.NewInternalError(err.Error())
+		var (
+			data  []byte
+			stale bool
+		)
+		if transformDecode {
+			var err error
+			if data, stale, err = s.transformer.TransformFromStorage(ctx, getResp.Kvs[0].Value, authenticatedDataString(key)); err != nil {
+				return nil, storage.NewInternalError(err.Error())
+			}
```

Comment:
why not move that block down behind line 926? Then you don't need two if clauses plus state.